Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -933,16 +959,30 @@ private extension PurchasesOrchestrator {
                 return false
             }
 
-            callbacks[productIdentifier] = completion
+            callbacks[productIdentifier] = .init(completion)
             return true
         }
     }
 
     func getAndRemovePurchaseCompletedCallback(
         forTransaction transaction: StoreTransaction
     ) -> PurchaseCompletedBlock? {
-        return self.purchaseCompleteCallbacksByProductID.modify {
-            $0.removeValue(forKey: transaction.productIdentifier)
+        return self.purchaseCompleteCallbacksByProductID.modify { callbacks -> PurchaseCompletedBlock? in
+            guard let value = callbacks[transaction.productIdentifier] else { return nil }
+
+            if !transaction.hasKnownPurchaseDate || value.creationDate <= transaction.purchaseDate {
```

Comment:
I know I'm late here but maybe we could extract this condition to a function `transactionDateMatchCallbackDate`, `transactionMatchesCallback` or something like that? Just so it's easier to read